kong/plugins/method-restriction/handler.lua

```diff
@@ -0,0 +1,32 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local responses = require "kong.tools.responses"
+local utils = require "kong.tools.utils"
+
+local MethodRestrictionHandler = BasePlugin:extend()
+
+MethodRestrictionHandler.PRIORITY = 940
+
+function MethodRestrictionHandler:new()
+  MethodRestrictionHandler.super.new(self, "method-restriction")
+end
+
+function MethodRestrictionHandler:access(conf)
+  MethodRestrictionHandler.super.access(self)
+
+  local block = false
+  local method = ngx.req.get_method()
+
+  if conf.blacklist and #conf.blacklist > 0 then
+    block = utils.table_contains(conf.blacklist, method)
+  end
+
+  if conf.whitelist and #conf.whitelist > 0 then
+    block = not utils.table_contains(conf.whitelist, method)
```

Comment:
About the cache, for this case, it's necessary create a reverse table iterating on it. I'm right?